concent_api/core/subtask_helpers.py

```diff
@@ -66,7 +66,25 @@ def update_timed_out_subtasks(
                 queue               = PendingResponse.Queue.ReceiveOutOfBand,
                 subtask             = subtask,
             )
-        assert subtask.state is not Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name  # pylint: disable=no-member
+        elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name:  # pylint: disable=no-member
+            # TODO: Make payment from requestor's deposit just like in the forced acceptance use case.
+            locked_subtask = Subtask.objects.select_for_update().get(pk=subtask.pk)
```

Comment:
I am not sure if I understand why we use `select_for_update` just here. Could you please explain?

Reply:
@dybi According to issue description

> Use database locks on the Subtask row corresponding to the task being processed to avoid race conditions between simultaneous requests or a request handler and a worker.

Reply:
We do need locks in more places. That's one I stated explicitly in the issue but I have more issues about that in the pipeline. If you see another place here were we would need a lock, it can be added too.